### ISSUE TITLE
`dvc exp pull <refname>` error: ERROR: unexpected error - 'str' object has no attribute 'baseline_sha'

### ISSUE THREAD

**Issue by patzy** [iterative/dvc#6421]
# Bug Report

## Issue name

When there are multiple experiments pushed to a git remote with the same name, trying to use `dvc pull` instructs to use the full refname to decide which one to pull. Doing so outputs an error, although the experiment seems to have been pulled correctly.

## Description

For example:
```console
$ dvc exp pull origin RandomForest
ERROR: Ambiguous name 'RandomForest' refers to multiple experiments in '{git_remote}'. Use full refname to pull one of the following:

	refs/exps/0a/57c2dab7898260bfdecce271f6b7f9cc22cd6b/RandomForest
	refs/exps/8e/96906ff946d98335625c8525d0dc0b54474665/RandomForest
```

Doing so prints an error:
```console
$ dvc exp pull origin refs/exps/0a/57c2dab7898260bfdecce271f6b7f9cc22cd6b/RandomForest
ERROR: unexpected error - 'str' object has no attribute 'baseline_sha'                                                                                                                         
```

But when listing experiments it seems it has been pulled correctly:
```console
$ dvc exp list --all
0a57c2d:                                                              
	RandomForest
```

### Reproduce

1. Create multiple experiments with the same name (this is possible by doing a git commit between experiments).
2. Push experiments as they're created
3. Clone the repo and attempt to pull a named experiment (since there are multiple ones with the same name you'll get the Ambiguous name error and have to pull by refname).
4. Pull experiment using refname

### Expected

There should be no error message ?

Also maybe make the doc more clear about this kind of edge cases ? (Seems to me we should not be able to create another experiment with the same name like this ?)

### Environment information

**Output of `dvc doctor`:**

```console
$ dvc doctor
DVC version: 2.5.4 (pip)
---------------------------------
Platform: Python 3.9.6 on Linux-5.13.7-arch1-1-x86_64-with-glibc2.33
Supports:
	http (requests = 2.26.0),
	https (requests = 2.26.0),
	s3 (s3fs = 2021.6.1, boto3 = 1.17.49),
	ssh (paramiko = 2.7.2)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sdb1
Caches: local
Remotes: s3
Workspace directory: ext4 on /dev/sdb1
Repo: dvc, git
```

**Additional Information (if any):**

Full verbose output of the command:
```console
$ dvc exp pull -v origin refs/exps/0a/57c2dab7898260bfdecce271f6b7f9cc22cd6b/RandomForest
2021-08-13 14:00:00,621 DEBUG: Check for update is enabled.
2021-08-13 14:00:00,623 DEBUG: git pull experiment 'origin' -> 'refs/exps/0a/57c2dab7898260bfdecce271f6b7f9cc22cd6b/RandomForest:refs/exps/0a/57c2dab7898260bfdecce271f6b7f9cc22cd6b/RandomForest'
2021-08-13 14:00:01,953 ERROR: unexpected error - 'str' object has no attribute 'baseline_sha'                                                                                                 
------------------------------------------------------------
Traceback (most recent call last):
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/main.py", line 55, in main
    ret = cmd.do_run()
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/command/base.py", line 50, in do_run
    return self.run()
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/command/experiments.py", line 682, in run
    self.repo.experiments.pull(
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/experiments/__init__.py", line 902, in pull
    return pull(self.repo, *args, **kwargs)
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/__init__.py", line 51, in wrapper
    return f(repo, *args, **kwargs)
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/scm_context.py", line 14, in run
    return method(repo, *args, **kw)
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/experiments/pull.py", line 35, in pull
    _pull_cache(repo, exp_ref, **kwargs)
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/experiments/pull.py", line 66, in _pull_cache
    revs = list(exp_commits(repo.scm, [exp_ref]))
  File "/home/m0rg/.local/lib/python3.9/site-packages/dvc/repo/experiments/utils.py", line 88, in exp_commits
    shas.update(scm.branch_revs(str(ref_info), ref_info.baseline_sha))
AttributeError: 'str' object has no attribute 'baseline_sha'
------------------------------------------------------------
2021-08-13 14:00:02,198 DEBUG: Version info for developers:
DVC version: 2.5.4 (pip)
---------------------------------
Platform: Python 3.9.6 on Linux-5.13.7-arch1-1-x86_64-with-glibc2.33
Supports:
	http (requests = 2.26.0),
	https (requests = 2.26.0),
	s3 (s3fs = 2021.6.1, boto3 = 1.17.49),
	ssh (paramiko = 2.7.2)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/sdb1
Caches: local
Remotes: s3
Workspace directory: ext4 on /dev/sdb1
Repo: dvc, git

Having any troubles? Hit us up at https://dvc.org/support, we are always happy to help!
2021-08-13 14:00:02,200 DEBUG: Analytics is enabled.
2021-08-13 14:00:02,235 DEBUG: Trying to spawn '['daemon', '-q', 'analytics', '/tmp/tmpl5_pxqxa']'
2021-08-13 14:00:02,237 DEBUG: Spawned '['daemon', '-q', 'analytics', '/tmp/tmpl5_pxqxa']'
```

**Comment by isidentical** [comment]
CC: @pmrowla

**Comment by pmrowla** [comment]
https://github.com/iterative/dvc/blob/13c1314099df131f526177b2a75bda02dfc0cdbf/dvc/repo/experiments/pull.py#L39-L40

this should be returning `ExpRefInfo.from_ref(exp_name)` instead of the string `exp_name`

**Comment by karajan1001** [comment]
Will be fixed after #6471
